### ISSUE TITLE
Possible lower than 32 bit??

### ISSUE THREAD

**Issue by kedaitinh12** [WolframRhodium/VapourSynth-BM3DCUDA#13]
Your plugin very fast but only if you use single bm3d. When bm3d use to mask for other denoiser (other denoiser only have cpu ver), it's very slow cause bm3d only support 32 bit. If you add support other bits lower than 32 bit, i think it's will faster

**Comment by WolframRhodium** [comment]
Thanks for the suggestion, I'll implement it in the next version.

**Comment by WolframRhodium** [comment]
Is `ConvertBits` slow? Implementing BM3DCPU in fixed point should be beneficial, but that's a lot of work for me.

**Comment by kedaitinh12** [comment]
Don't really for convertbits. I just talk about bm3d will mask noise pixel and other denoiser plugins will work on mask noise pixel by bm3d. Example: bm3dGPU work faster than KNLMEANSCL when use them all alone. But when use with TemporalDegrain2 postFFT=4 (neo-fft3d work with KNLMEANSCL) work faster than postFFT=5 (neo-fft3d work with BM3DGPU) cause postFFT=5 need convertbit(32) for work. Neo-fft3d work in 16 bit (with KNLMEANSCL) faster than 32 bit (with BM3DGPU).

**Comment by kedaitinh12** [comment]
> Is `ConvertBits` slow? Implementing BM3DCPU in fixed point should be beneficial, but that's a lot of work for me.

If you don't have time i will close issue. Thanks

**Comment by WolframRhodium** [comment]
The issue should be addressed in this case. I will take a look on it.

**Comment by WolframRhodium** [comment]
The issue should be addressed in this case. I will take a look at it.

**Comment by WolframRhodium** [comment]
Have you tried the [z.lib resizers](https://forum.doom9.org/showthread.php?t=173986)? It seems `ConvertBits` does not has vectored subroutines for uint16 -> fp32 conversion.

**Comment by kedaitinh12** [comment]
I only use convertbit and zlib need convertformat to use convertbit

**Comment by WolframRhodium** [comment]
Yes, `z_ConvertFormat` should be faster than `convertbit` in terms of bitdepth conversion.

**Comment by kedaitinh12** [comment]
But i think BM3DGPU able to use 16 bit or lower with neo_fft3d will faster than KNLMEANSCL in same bit
